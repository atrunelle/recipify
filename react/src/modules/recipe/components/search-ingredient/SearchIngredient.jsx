import './SearchIngredient.css';

import React, { Component } from 'react';
import { Typography, TextField, Button, Card, CardContent, Grid, CircularProgress } from '@material-ui/core';

class SearchIngredient extends Component {
  constructor(props) {
    super(props);

    this.state = {
      ingredientName: '',
      numberOfServings: 1,
    };
  }

  onIngredientChange = (event) => {
    this.setState({ ingredientName : event.target.value });
  }

  onNumberOfServingsChange = (event) => {
    this.setState({ numberOfServings : event.target.value });
  }

  onSubmit = (event) => {
    event.preventDefault();
    this.props.add(this.state.ingredientName, this.state.numberOfServings);
  }
  
  render() {
    let buttonContent = '';

    if (this.props.isFetching) {
      buttonContent = <CircularProgress style={{ color: '#fff' }} />;
    } else {
      buttonContent = 'Add ingredient';
    }

    return (
      <Card className="u-push-top">
        <CardContent>
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            What do you have in your fridge?
          </Typography>          
          <form className="search" onSubmit={this.onSubmit}>
            <Grid 
              container
              spacing={8}
              justify="space-between">
              <Grid
                item
                xs={12} 
                sm={6}>
                <TextField
                  fullWidth
                  label="Find an ingredient"
                  value={this.state.ingredientName} 
                  id="ingredientNameInput"
                  placeholder="Ex: 200gr chicken"
                  onChange={this.onIngredientChange}/>
              </Grid>
              <Grid item>
                <TextField
                  label="Number of servings"
                  value={this.state.numberOfServings}
                  id="numberOfServingsInput"
                  onChange={this.onNumberOfServingsChange}/>
              </Grid>
              <Button
                disabled={!this.state.ingredientName}
                type="submit"
                variant="raised"
                color="primary"
                id="submitButton">
                {buttonContent}
              </Button>
            </Grid>
          </form>
        </CardContent>
      </Card>
    );
  }
}

export default SearchIngredient;
