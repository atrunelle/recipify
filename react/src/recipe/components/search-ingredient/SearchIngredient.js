import './SearchIngredient.css';

import React, { Component } from 'react';

class SearchIngredient extends Component {
    constructor(props) {
        super(props);
    
        this.state = {
          ingredientName: '',
          numberOfServings: 1,
        };
    }

    onIngredientChange = (event) => {
      this.setState({ ingredientName : event.target.value });
    }

    onNumberOfServingsChange = (event) => {
      this.setState({ numberOfServings : event.target.value });
    }

    onSubmit = (event) => {
      event.preventDefault();
      this.props.add(this.state.ingredientName, this.state.numberOfServings);
    }
    
  render() {
    return (
        <form className="search" onSubmit={this.onSubmit}>
          <label className="search__label">
            What do you want to use?
          </label>
          <input 
            className="search__input" 
            value={this.state.ingredientName} 
            type="text" 
            placeholder="Find an ingredient..."
            onChange={this.onIngredientChange}/>
          <input 
            className="search__input" 
            value={this.state.numberOfServings} 
            type="text" 
            placeholder="Find an ingredient..."
            onChange={this.onNumberOfServingsChange}/>
            <button type="submit">Submit</button>
        </form>
    );
  }
}

export default SearchIngredient;
