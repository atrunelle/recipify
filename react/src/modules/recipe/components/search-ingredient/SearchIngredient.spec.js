import React from 'react';
import { mount } from 'enzyme';
import SearchIngredient from './SearchIngredient';

describe('Component: SearchIngredient', () => {
  let component;

  const mockAdd = jest.fn();
  const mockIsFetching = false;

  beforeEach(() => {
    component = mount(
      <SearchIngredient
        add={mockAdd}
        isFetching={mockIsFetching}/>
    );
  });
  
  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});