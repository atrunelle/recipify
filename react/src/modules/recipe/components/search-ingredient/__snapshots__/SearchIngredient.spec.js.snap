// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchIngredient should match snapshot 1`] = `"<div class=\\"MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1 u-push-top\\"><div class=\\"MuiCardContent-root-29\\"><h2 class=\\"MuiTypography-root-30 MuiTypography-headline-35 MuiTypography-gutterBottom-47\\">What do you have in your fridge?</h2><form class=\\"search\\"><div class=\\"MuiGrid-container-54 MuiGrid-spacing-xs-8-75 MuiGrid-justify-xs-space-between-73\\"><div class=\\"MuiGrid-item-55 MuiGrid-grid-xs-12-92 MuiGrid-grid-sm-6-99\\"><div class=\\"MuiFormControl-root-145 MuiFormControl-fullWidth-148\\"><label class=\\"MuiFormLabel-root-154 MuiInputLabel-root-149 MuiInputLabel-formControl-150 MuiInputLabel-animated-153\\" data-shrink=\\"false\\" for=\\"ingredientNameInput\\">Find an ingredient</label><div class=\\"MuiInput-root-159 MuiInput-fullWidth-166 MuiInput-formControl-160 MuiInput-underline-163\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-167\\" id=\\"ingredientNameInput\\" placeholder=\\"Ex: 200gr chicken\\" value=\\"\\"></div></div></div><div class=\\"MuiGrid-item-55\\"><div class=\\"MuiFormControl-root-145\\"><label class=\\"MuiFormLabel-root-154 MuiInputLabel-root-149 MuiInputLabel-formControl-150 MuiInputLabel-animated-153 MuiInputLabel-shrink-152\\" data-shrink=\\"true\\" for=\\"numberOfServingsInput\\">Number of servings</label><div class=\\"MuiInput-root-159 MuiInput-formControl-160 MuiInput-underline-163\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-167\\" id=\\"numberOfServingsInput\\" value=\\"1\\"></div></div></div><button tabindex=\\"-1\\" class=\\"MuiButtonBase-root-188 MuiButtonBase-disabled-189 MuiButton-root-172 MuiButton-raised-178 MuiButton-raisedPrimary-179 MuiButton-disabled-182\\" type=\\"submit\\" disabled=\\"\\" id=\\"submitButton\\"><span class=\\"MuiButton-label-173\\">Add ingredient</span></button></div></form></div></div>"`;
